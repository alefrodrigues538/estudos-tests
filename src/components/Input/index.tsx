import { FC } from "react";

interface InputProps {
  value: string | number;
  placeholder?: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  type?: "text" | "number" | "email" | "password";
}

export function handleChange(
  event: React.ChangeEvent<HTMLInputElement>,
  onChange: (value: any) => void
) {
  onChange(event.target.value);
}

const Input: FC<InputProps> = ({ value, placeholder, onChange, type }) => {
  return (
    <input
      type={type}
      value={value}
      placeholder={placeholder}
      onChange={onChange}
    />
  );
};

export default Input;
