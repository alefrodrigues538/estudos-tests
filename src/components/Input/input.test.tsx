/* eslint-disable testing-library/render-result-naming-convention */
/* eslint-disable testing-library/no-node-access */
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import React from "react";
import Input, { handleChange } from "./index";

describe("Input", () => {
  test("Should be render a Input with attributes", () => {
    const onChangeMock = jest.fn();
    const { container } = render(
      <Input
        type="text"
        value={"input"}
        placeholder="Digite aqui..."
        onChange={onChangeMock}
      />
    );

    const inputElement = screen.getByPlaceholderText("Digite aqui...");

    expect(container.firstChild).toMatchSnapshot();
    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveAttribute("type", "text");
  });

  test("Should be call onChange with a brand new value", () => {
    const onChange = jest.fn();

    const event = {
      target: { value: "10" },
    } as React.ChangeEvent<HTMLInputElement>;
    handleChange(event, onChange);

    expect(onChange).toHaveBeenCalledWith("10");
  });
});
