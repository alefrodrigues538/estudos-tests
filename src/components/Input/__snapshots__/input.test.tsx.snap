// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should be render a Input with attributes 1`] = `
<input
  placeholder="Digite aqui..."
  type="text"
  value="input"
/>
`;
